unit_tests/test_aggregator.py

```diff
@@ -0,0 +1,434 @@
+        aggregator_explicit._connect_to_simulation.assert_called_once()
+
+    @staticmethod
+    def test_constructor_latest_grid_tree_setup(aggregator_explicit):
+        assert aggregator_explicit.latest_grid_tree == {}
+
+    @staticmethod
+    def test_constructor_latest_grid_tree_flat_setup(aggregator_explicit):
+        assert aggregator_explicit.latest_grid_tree_flat == {}
+
+    @staticmethod
+    def test_constructor_area_name_uuid_mapping_setup(aggregator_explicit):
+        assert aggregator_explicit.area_name_uuid_mapping == {}
+
+
+class TestAggregatorStartWebsocketConnection:
+    """Test cases for Aggregator's start_websocket_connection method."""
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_start_websocket_funcs")
+    def test_start_websocket_connection_dispatcher_instantiated(aggregator_explicit):
+        aggregator_explicit.start_websocket_connection()
+        assert aggregator_explicit.dispatcher is not None
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_start_websocket_funcs")
+    def test_start_websocket_connection_websocket_instantiated(aggregator_explicit):
+        aggregator_explicit.aggregator_uuid = TEST_AGGREGATOR_UUID
+        aggregator_explicit.start_websocket_connection()
+        assert aggregator_explicit.websocket_thread is not None
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_start_websocket_funcs")
+    def test_start_websocket_connection_websocket_thread_start(aggregator_explicit):
+        aggregator_explicit.start_websocket_connection()
+        aggregator_explicit.websocket_thread.start.assert_called_once()
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_start_websocket_funcs")
+    def test_start_websocket_connection_callback_thread_instantiated(aggregator_explicit):
+        aggregator_explicit.start_websocket_connection()
+        assert aggregator_explicit.callback_thread is not None
+
+
+class TestAggregatorGetUuidFromAreaName:
+    """Test cases for Aggregator's get_uuid_from_area_name method."""
+
+    @staticmethod
+    def test_get_uuid_from_area_name_returns_expected_with_non_empty_dict(aggregator_explicit):
+        aggregator_explicit.area_name_uuid_mapping = TEST_AREA_MAPPING
+        area_name = TEST_AREA_NAME
+        ret_val = TEST_AREA_MAPPING[TEST_AREA_NAME][0]
+
+        patch("gsy_e_sdk.aggregator.get_uuid_from_area_name_in_tree_dict",
+              return_value=ret_val)
+        assert aggregator_explicit.get_uuid_from_area_name(area_name) == ret_val
+
+    @staticmethod
+    def test_get_uuid_from_area_name_returns_none_with_empty_dict(aggregator_explicit):
+        aggregator_explicit.area_name_uuid_mapping = {}
+        area_name = TEST_AREA_NAME
+        ret_val = TEST_AREA_MAPPING[TEST_AREA_NAME][0]
+
+        patch("gsy_e_sdk.aggregator.get_uuid_from_area_name_in_tree_dict",
+              return_value=ret_val)
+        assert aggregator_explicit.get_uuid_from_area_name(area_name) is None
```

Comment:
NIT, this can be simplified and combined with the test above (`test_get_uuid_from_area_name_returns_expected_with_non_empty_dict`) using pyrametrize.
(please find better names for the variables in my propsoal:
```suggestion
@pytest.mark.parametrize("name_uuid_mapping, expected_ret_val", [[TEST_AREA_MAPPING, {}], [TEST_AREA_MAPPING[TEST_AREA_NAME][0], None]])
    def test_get_uuid_from_area_name_returns_none_with_empty_dict(name_uuid_mapping, expected_ret_val, aggregator_explicit):
        aggregator_explicit.area_name_uuid_mapping = name_uuid_mapping
        patch("gsy_e_sdk.aggregator.get_uuid_from_area_name_in_tree_dict",
              return_value=TEST_AREA_MAPPING[TEST_AREA_NAME][0])
        assert aggregator_explicit.get_uuid_from_area_name(TEST_AREA_NAME) is expected_ret_val
```

Reply:
I addressed it. The decorator should look like this though
```
@pytest.mark.parametrize("uuid_mapping, expected_ret_val",
                         [(TEST_AREA_MAPPING, TEST_AREA_MAPPING[TEST_AREA_NAME][0]), ({}, None)])
```

The class containing these previous test cases (`TestAggregatorGetUuidFromAreaName`) had then only one test case parametrized with these two possibilities. Therefore I deleted the class and moved the test at the end of the module to be tested with the "individual" ones.